### PR TITLE
Perf compiler typecheck

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: packages/compiler-cli/src/transformers/api.ts [hunk 1]
```diff
@@ -21,6 +21,14 @@ export interface Diagnostic {
   source: 'angular';
 }
 
+export function isTsDiagnostic(diagnostic: any): diagnostic is ts.Diagnostic {
+  return diagnostic && diagnostic.source != 'angular';
```
Comment (vicb) [inline packages/compiler-cli/src/transformers/api.ts]: should return a boolean (same below)

Comment (vicb) [reply packages/compiler-cli/src/transformers/api.ts]: also use `!==` for consistency while you're at it

File: packages/compiler/src/view_compiler/type_check_compiler.ts [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+  private outputVarTypes = new Map<string, OutputVarType>();
+  private outputVarNames = new Map<OutputVarType, string>();
+  private refOutputVars = new Map<string, OutputVarType>();
+  private variables: VariableAst[] = [];
+  private children: ViewBuilder[] = [];
+  private updates: Expression[] = [];
+  private actions: Expression[] = [];
+
+  constructor(
+      private _options: AotCompilerOptions, private reflector: StaticReflector,
```
Comment (vicb) [inline packages/compiler/src/view_compiler/type_check_compiler.ts]: nit: why `_` only for one field ? remove ?

**Review by vicb — APPROVED** [review]
Please address the minor comments

File: packages/compiler-cli/src/transformers/api.ts (hunk 1 above)
Comment (tbosch) [reply packages/compiler-cli/src/transformers/api.ts]: Done

File: packages/compiler/src/view_compiler/type_check_compiler.ts (hunk 2 above)
Comment (tbosch) [reply packages/compiler/src/view_compiler/type_check_compiler.ts]: Removed all `_` in this file...

**Pull request closed without merging**